src/client/shader.cpp

```diff
@@ -620,15 +653,58 @@ ShaderInfo generate_shader(const std::string &name, u8 material_type, u8 drawtyp
 		return shaderinfo;
 
 	// Create shaders header
-	std::string shaders_header = "#version 120\n";
-
+	bool use_gles = false;
+#if ENABLE_GLES
+	use_gles = driver->getDriverType() == video::EDT_OGLES2;
+#endif
+	std::string shaders_header, vertex_header, pixel_header; // geometry shaders aren’t supported in GLES<3
+	if (use_gles) {
+		shaders_header = \
+			"#version 100\n" \
+			"varying mediump vec4 varTexCoord[1];\n" \
+			"#define gl_TexCoord varTexCoord\n" \
+			;
+		vertex_header = \
+				"uniform highp mat4 mWorldView;\n" \
+				"uniform mat4 mTexture[1];\n" \
+				"uniform mat3 mNormal;\n" \
+				"attribute highp vec4 inVertexPosition;\n" \
+				"attribute vec3 inVertexNormal;\n" \
+				"attribute vec4 inVertexColor;\n" \
+				"attribute vec4 inTexCoord0;\n" \
+	/* 			"attribute vec4 inTexCoord1;\n" */\
+				"attribute vec4 inVertexTangent;\n" \
+				"attribute vec4 inVertexBinormal;\n" \
+				"\n" \
+				"#define gl_ModelViewMatrix mWorldView\n" \
+				"#define gl_TextureMatrix mTexture\n" \
+				"#define gl_NormalMatrix mNormal\n" \
+				"#define gl_Vertex inVertexPosition\n" \
```

Comment:
All the `\`'s are not necessary here

Reply:
and you don’t mind redefining the reserved words...

Reply:
those aren't reserved in GLES's GLSL are they?

Reply:
>  Identifiers starting with “gl_” are reserved for use by OpenGL ES.  No user-defined identifiers may begin with “gl_”.